### PR TITLE
Improved Packet version mismatch exception message

### PR DESCRIPTION
**Pull request by pveentjer**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/nio/Packet.java [hunk 1]
```diff
@@ -188,8 +188,11 @@ private boolean readVersion(ByteBuffer source) {
             byte version = source.get();
             setPersistStatus(PERSIST_VERSION);
             if (VERSION != version) {
-                throw new IllegalArgumentException("Packet versions are not matching! Expected -> "
-                        + VERSION + ", Incoming -> " + version);
+                throw new IllegalArgumentException(
+                        "Packet version mismatch; the incoming packet version doesn't match expected packet version,"
+                                + " incoming packet version: " + version
+                                + " expected packet version: " + VERSION + '.'
+                                + " If the incoming version is negative or very large, it could indicate a corrupted stream.");
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/nio/Packet.java]: just an idea: what about including this only if the incoming is negative or large ?

**Comment by jerrinot** [comment]
Good One! :+1:

File: hazelcast/src/main/java/com/hazelcast/nio/Packet.java (hunk 1 above)
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/nio/Packet.java]: +1

I'll play with that a bit.

Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/nio/Packet.java]: perhaps you could give a hint: There is a different version of Hazelcast running.

Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/nio/Packet.java]: @pveentjer: ping

**Pull request closed without merging**